cpp/fregrid/bilinear_interp.C

```diff
@@ -134,79 +144,84 @@ void setup_bilinear_interp(int ntiles_in, const Grid_config *grid_in, int ntiles
-  found    = (int *)malloc(nx_out*ny_out*sizeof(int));
-  index    = (int *)malloc(ntiles_in*3*sizeof(int));
-  shortest = (double *)malloc(ntiles_in*sizeof(double));
-  for(i=0; i<nx_out*ny_out; i++) found[i] = 0;
-
-  dlon = dlon_in/nx_out;
-  dlat = dlat_in/(ny_out-1);
-
-  for(iter=1; iter<=max_iter; iter++) {
-    for(l=0; l<ntiles_in; l++) {
-      for(jc=1; jc<=ny_in; jc++) for(ic=1; ic<=nx_in; ic++) {
-	/*------------------------------------------------------
-	  guess latlon indexes for given cubed sphere cell
-	  ------------------------------------------------------*/
-	n1 = jc*nxd+ic;
-	n2 = (jc+1)*nxd+ic+1;
-	v1[0] = grid_in[l].xt[n1];
-	v1[1] = grid_in[l].yt[n1];
-	v1[2] = grid_in[l].zt[n1];
-	v2[0] = grid_in[l].xt[n2];
-	v2[1] = grid_in[l].yt[n2];
-	v2[2] = grid_in[l].zt[n2];
-	dcub=iter*normalize_great_circle_distance(v1, v2);
-	j_min=max( 1,  static_cast<int>(floor((grid_in[l].latt[n1]-dcub-latbegin)/dlat)-iter+1)); //TODO: possible error
-	j_max=min(ny_out,static_cast<int>(ceil((grid_in[l].latt[n1]+dcub-latbegin)/dlat)+iter-1));//TODO: possible error
-	if (j_min==1 || j_max==ny_out) {
-	  i_min=1;
-	  i_max=nx_out;
-	}
-        else {
-	  i_min=max(   1, static_cast<int>( floor((grid_in[l].lont[n1]-dcub-lonbegin)/dlon-iter+1)));
-	  i_max=min(nx_out,static_cast<int>(ceil((grid_in[l].lont[n1]+dcub-lonbegin)/dlon+iter-1)));
-	}
-	for(j=j_min-1; j<j_max; j++) for(i=i_min-1; i<i_max; i++) {
-	  n0 = j*nx_out + i;
-	  /*--------------------------------------------------------------------
-            for latlon cell find nearest cubed sphere cell
-            ------------------------------------------------------------------*/
-	  if (!found[n0]) {
-	    shortest[l]=M_PI+M_PI;
-	    for(jcc=jc; jcc<=min(ny_in, jc+1); jcc++) for(icc=ic; icc<=min(nx_in, ic+1); icc++) {
-	      n1 = jcc*nxd + icc;
-	      n2 = j*nx_out + i;
-	      v1[0] = grid_in[l].xt[n1];
-	      v1[1] = grid_in[l].yt[n1];
-	      v1[2] = grid_in[l].zt[n1];
-	      v2[0] = grid_out->xt[n2];
-	      v2[1] = grid_out->yt[n2];
-	      v2[2] = grid_out->zt[n2];
-	      distance=normalize_great_circle_distance(v1, v2);
-	      if (distance < shortest[l]) {
-		shortest[l]=distance;
-		index[3*l]  =icc;
-		index[3*l+1]=jcc;
-		index[3*l+2]=l;
-	      }
-	    }
-	    /*------------------------------------------------
-              determine lower left corner
-              ------------------------------------------------*/
-	    found[n0] = get_closest_index(&(grid_in[l]), grid_out, &(interp->index[3*(j*nx_out+i)]), index[3*l],
-			      index[3*l+1], index[3*l+2], i, j);
-	  }
-	}
-      }
+  found = (int *) malloc(nx_out * ny_out * sizeof(int));
+  index = (int *) malloc(ntiles_in * 3 * sizeof(int));
+  shortest = (double *) malloc(ntiles_in * sizeof(double));
+  for (i = 0; i < nx_out * ny_out; i++) found[i] = 0;
+
+  dlon = dlon_in / nx_out;
+  dlat = dlat_in / (ny_out - 1);
+
+  for (iter = 1; iter <= max_iter; iter++) {
+    for (l = 0; l < ntiles_in; l++) {
```

Comment:
Small nit-pick, but you did ask for a review, so... 

obviously one can tell `l` from `1` via syntax highlighting... but it's not exactly fun to have to rely on the syntax highlighter nor a few pixels of difference either.

But, i have a feeling `l` is used as a general index in multiple places, and so constitutes consistent style. It is also generally consistent with vector/tensor component notation. 

This isn't something that needs to be resolved here, but i want to comment this here now in case such a thing does become a headache for some reason in the future.